### PR TITLE
Add the ability to ignore files.

### LINKED ISSUE #11: Add flag to pass in filenames to leave unmodified
I ran polyup against this project https://github.com/christopheclc/polymer-date-picker/ as a test, and it behaves a tad oddly.
```
polyup polymer-date-picker.html
```

1) if no --overwrite is given it inlines moment.min.js as well...
2) if --overwrite is given it aborts because of moment.min.js being outside CWD (understandable precaution but, really it should ignore it right?

### PR DESCRIPTION
**Pull request by rictic**
Fixes #11

### CODE REVIEW

File: lib/node_main.js [hunk 1]
```diff
@@ -13,49 +13,62 @@
...
+var filename = opts.path_to_html;
 if (!fs.existsSync(filename)) {
   console.error('No such file: ' + filename);
   process.exit(66);
 }
 
+var setToIgnore = new Set();
+opts.ignore.forEach(function(toIgnore) {
+  toIgnore = path.resolve(toIgnore);
+  if (!fs.existsSync(toIgnore)) {
```
Comment (justinfagnani) [inline lib/node_main.js]: does it really matter if an ignored file doesn't exist?

File: lib/node_main.js [hunk 2]
```diff
@@ -13,49 +13,62 @@
...
+    console.error(
+        'No file exists at path `' + toIgnore + '`');
```
Comment (justinfagnani) [inline lib/node_main.js]: looks like this will fit on one line

File: lib/node_main.js (hunk 1 above)
Comment (rictic) [inline lib/node_main.js]: My thinking was that it's probably a typo, so you'll get unexpected behavior when the file you tried to ignore is actually overwritten.

File: lib/node_main.js [hunk 3]
```diff
@@ -13,49 +13,62 @@
...
-    .argv;
-
-var filename = argv._[0];
+var filename = opts.path_to_html;
 if (!fs.existsSync(filename)) {
   console.error('No such file: ' + filename);
   process.exit(66);
 }
 
+var setToIgnore = new Set();
```
Comment (justinfagnani) [inline lib/node_main.js]: nit: maybe something like `filesToIgnore`? `set` encodes the type in the name.

File: lib/upgrade_html.js [hunk 4]
```diff
@@ -253,14 +256,17 @@ function upgradeHtml(filename) {
...
+function upgradeScriptElement($, docFilename, scriptElem, toIgnore, attrs,
+                              hostAttrs, elemName, newDeclarations) {
   if ('src' in scriptElem.attribs) {
     var srcPath = scriptElem.attribs.src;
     if (PathResolver.prototype.isAbsoluteUrl(srcPath)) {
       return;
     }
     var pathToScriptElem = path.resolve(path.dirname(docFilename), srcPath);
+    if (toIgnore.has(pathToScriptElem)) {
+      return;
```
Comment (justinfagnani) [inline lib/upgrade_html.js]: are scripts the only thing you need to ignore?

File: lib/node_main.js (hunk 3 above)
Comment (rictic) [inline lib/node_main.js]: Good point. How about just toIgnore, to match how it's named in upgrade_html.js

File: lib/upgrade_html.js (hunk 4 above)
Comment (rictic) [inline lib/upgrade_html.js]: Yeah, I looked for all other uses of the fs module.

File: lib/node_main.js (hunk 2 above)
Comment (rictic) [inline lib/node_main.js]: Done.

**Comment by rictic** [comment]
All comments addressed, ptal

File: lib/node_main.js (hunk 1 above)
Comment (justinfagnani) [inline lib/node_main.js]: good point

File: lib/node_main.js (hunk 3 above)
Comment (justinfagnani) [inline lib/node_main.js]: sgtm

**Comment by justinfagnani** [comment]
LGTM

File: lib/node_main.js (hunk 3 above)
Comment (dave-2) [inline lib/node_main.js]: Looks like this introduced a bug. toIgnore now has a naming conflict between the set of files and the individual file to ignore. I'll send a pull request.

```
/usr/local/lib/node_modules/polyup/dist/node_main.js:106
  toIgnore.add(toIgnore);
           ^
TypeError: Object dashboard/elements/base-chart.html has no method 'add'
```

**Pull request merged**